app/src/main/java/com/arctouch/codechallenge/util/ResultAsync.kt

```diff
@@ -0,0 +1,45 @@
+package com.arctouch.codechallenge.util
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.android.Main
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class ResultAsync<T> private constructor(action: suspend () -> T, scope: CoroutineScope) {
```

Comment:
Very nice usage of high-order functions and extension functions! 👏

Reply:
Thanks, I was inspired on Kotlin Result Class that returns when you use runCatching, but is only to synchronous code, I adapted it to use coroutines with the same pattern.